include/bout/solver.hxx

```diff
@@ -324,6 +324,13 @@ public:
   /// The run from which this was restarted. Throws if the identifier hasn't been set yet.
   std::string getRunRestartFrom() const;
 
+  /// Get the number of completed output steps
+  int getIterationCounter() const { return iteration; }
+
+  /// Add one to the iteration count, used by BoutMonitor, but could be called by a
+  // user-defined monitor (if `bout_run()` is not used)
+  int incrementIterationCounter() { return iteration++; }
```

Comment:
This returns the value before the increment, but I think we probably want the value after increment:

```suggestion
  int incrementIterationCounter() { return ++iteration; }
```

Otherwise we're back to the original situation? Is it worth adding a test, perhaps to `test-restarting`, that checks `hist_hi` and `iteration`?

It is _incredibly_ unfortunate that `++` _after_ the variable returns the value _before_ the increment, and vice versa.

Reply:
Yes, absolutely! Sorry, I mis-remembered and thought `++i` and `i++` were the other way around. Thank you for spotting my error!

Reply:
I have to check every time, it's confusing! :smile: